Code/Framework/AzCore/Tests/Math/IntersectionTests.cpp

```diff
@@ -496,109 +496,109 @@ namespace UnitTest
         EXPECT_GT(numMisses, 0);
     }
 
-#if AZ_TRAIT_DISABLE_FAILED_MATH_TESTS
-    TEST_F(MATH_IntersectSegmentTriangleTest, DISABLED_RegressionTestForSpecificSegmentsAndTriangles)
-#else
-    TEST_F(MATH_IntersectSegmentTriangleTest, RegressionTestForSpecificSegmentsAndTriangles)
-#endif
-    {
-        // This unit test is set up to validate regressions of erroneous segment/triangle intersection test results.
-        // All the failures that have been found so far have been false negatives or false positives that occur on triangle
-        // edges or vertices.
+     struct RayTriangleTest
+    {
+        AZ::Vector3 m_rayStart;
+        AZ::Vector3 m_rayEnd;
+        AZ::Vector3 m_triVerts[3];
+        bool m_shouldHit;
+        float m_hitDistance;
+        AZ::Vector3 m_hitNormal;
+    };
 
-        struct RayTriangleTest
+    static const RayTriangleTest RayTriangleTestParams[] =
+    {
+        // This failure came from a vegetation system raycast to a triangle in a sphere model.
         {
-            AZ::Vector3 m_rayStart;
-            AZ::Vector3 m_rayEnd;
-            AZ::Vector3 m_triVerts[3];
-            bool m_shouldHit;
-            float m_hitDistance;
-            AZ::Vector3 m_hitNormal;
-        };
+            { 0.0, 0.16, 1.01948643 }, { 0.0, 0.16, -0.0194873810 }, // segment
```

Comment:
nit: While you're here could you add `f` to the floating point params to avoid the implicit double conversions please

Reply:
I'm `99.9e+36`% sure that's not going to break anything. But the last couple of days I've been fighting with precision issues witnessing very minor changes making these tests fail, so I'm going to play it safe and not change that for now.

Reply:
FWIW, there is no true implicit double conversion here. The "conversion" takes place at compile time, the compiler produces the same results whether or not the f is there (see godbolt to verify). I'm not opposed to someone adding the f, but it seems like purely a style choice. 

If we were going to modify these constants, my preference would be that we go a step further and actually write out the full exact values that are stored in the float instead of these lightly-rounded values, so that way anyone who wants to try doing the math by hand would be starting from the exact same numbers that the computer is using. That feels pretty far outside the range of this PR though.